member/src/main/proto/com/improving/app/member/domain/improving-app-member-domain.proto

```diff
@@ -0,0 +1,198 @@
+syntax =  "proto3";
+
+package com.improving.app.member.domain;
+
+import "com/improving/app/organization/api/improving-app-organization-api.proto";
+import "scalapb/scalapb.proto";
+
+option (scalapb.options) = {
+  no_default_values_in_constructor: true
+  preserve_unknown_fields: false
+};
+
+
+//Data Model
+message MemberId {
+  option (scalapb.message).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  string id = 1;
+}
+
+enum MemberType {
+  option (scalapb.enum_options).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  MEMBER_TYPE_STUDENT = 0;
+  MEMBER_TYPE_FACULTY_STAFF = 1;
+  MEMBER_TYPE_VENDOR = 2;
+  MEMBER_TYPE_SPONSOR =3;
+  MEMBER_TYPE_GENERAL = 4;
+  MEMBER_TYPE_ALUMNI = 5;
+  MEMBER_TYPE_PARENT = 6;
+  MEMBER_TYPE_FAN = 7;
+}
+
+enum NotificationPreference {
+  option (scalapb.enum_options).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  NOTIFICATION_PREFERENCE_EMAIL = 0;
+  NOTIFICATION_PREFERENCE_SMS = 1;
+}
+
+enum MemberStatus {
+  option (scalapb.enum_options).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  MEMBER_STATUS_INITIAL = 0;
+  MEMBER_STATUS_ACTIVE = 1;
+  MEMBER_STATUS_INACTIVE = 2;
+  MEMBER_STATUS_SUSPENDED = 3;
+  MEMBER_STATUS_TERMINATED = 4;
+}
```

Comment:
The presence of this means you're picking a handler based state machine mechanism rather than  state-based state machine. Your state changes will be with the "become" action, not the "morph" action. Just want you to be aware of that.

Reply:
Yes going for handler based approach

Reply:
The name convention for the proto file was decided to be using _ in all cases for Kalix study, i.e. improving_app_member_domain.proto.  I am not sure if we like to follow that here as well.

Reply:
I prefer - over _ in file names because only one key press is needed. :)

Reply:
agreed